### PR TITLE
Add protected tags functionality

### PR DESCRIPTION
**Pull request by therealgambo**

### CODE REVIEW

File: gitlabform/gitlabform/core.py [hunk 1]
```diff
@@ -307,6 +308,28 @@ def process_branches(self, project_and_group, configuration):
                 if self.args.strict:
                     exit(3)
 
+    @if_in_config_and_not_skipped
+    def process_tags(self, project_and_group, configuration):
+        for tag in sorted(configuration['tags']):
+            try:
+                if configuration['tags'][tag]['protected']:
+                    create_access_level = configuration['tags'][tag]['create_access_level'] if \
+                        'create_access_level' in configuration['tags'][tag] else '30'
```
Comment (gdubicki) [inline gitlabform/gitlabform/core.py]: I would not pass any value to GitLab's API if user won't set the `create_access_level` in gitlabform's config. Then GitLab's API default (`40`, according to https://docs.gitlab.com/ee/api/protected_tags.html#protect-repository-tags), will be used, which I think is what the user expects.

**Comment by gdubicki** [comment]
Thank you for your contributions, @therealgambo ! 🥇 

I did a review, please see my comments above and in https://github.com/egnyte/gitlabform/pull/40 .

**Comment by therealgambo** [comment]
@gdubicki updated with changes to only send that parameter if it is defined in the configuration.

File: gitlabform/gitlab/tags.py [hunk 2]
```diff
@@ -14,3 +14,16 @@ def create_tag(self, project_and_group_name, tag_name, ref, message=None):
         }
         return self._make_requests_to_api("projects/%s/repository/tags", project_and_group_name, method='POST', data=data,
                                           expected_codes=201)
+
+    def protect_tag(self, project_and_group_name, tag_name, create_access_level):
+        data = {
+            'name': tag_name
+        }
+        if create_access_level:
```
Comment (glensc) [inline gitlabform/gitlab/tags.py:22]: I think this is the reason for this bug:
- https://github.com/egnyte/gitlabform/issues/172

File: gitlabform/gitlabform/core.py (hunk 1 above)
Comment (glensc) [reply gitlabform/gitlabform/core.py]: the "exists"-check is done incorrectly:
- https://github.com/egnyte/gitlabform/pull/39#discussion_r566725404

**Pull request merged**